client/state/jetpack-connect/actions.js

```diff
@@ -31,12 +31,15 @@ import {
 } from 'state/action-types';
 import userFactory from 'lib/user';
 import config from 'config';
+import i18nUtils from 'lib/i18n-utils';
 
 /**
  *  Local variables;
  */
 let _fetching = {};
-const authURL = '/wp-admin/admin.php?page=jetpack&connect_url_redirect=true&calypso_env=' + config( 'env' );
+const authURL = '/wp-admin/admin.php?page=jetpack&connect_url_redirect=true' +
+	'&calypso_env=' + config( 'env' ) +
+	'&locale=' + i18nUtils.getLocaleSlug();
```

Comment:
Awesome, noted! I think we can drop `locale` from this, since we decided to handle it on the Jetpack side.